dlt-connector/src/client/IotaClientSingleton_disabled.test.ts

```diff
@@ -0,0 +1,26 @@
+import { IotaClientSingleton } from '@/client/IotaClientSingleton'
+import CONFIG from '@/config'
+import { logger } from '@/server/logger'
+
+describe('apis/IotaClientSingleton/disabled', () => {
+  beforeEach(() => {
+    CONFIG.IOTA = false
+  })
+  it('getInstance return undefined if iota is disabled', () => {
+    const spyLog = jest.spyOn(logger, 'info')
+    expect(IotaClientSingleton.getInstance()).toBeUndefined()
+    expect(spyLog).toHaveBeenCalledWith('Iota are disabled via config...')
+  })
+})
+
+describe('apis/IotaClientSingleton/invalidIotaUrl', () => {
+  beforeEach(() => {
+    CONFIG.IOTA = true
+    CONFIG.IOTA_API_URL = 'invalidUrl'
+  })
+  it('log "couldn\'t connect to iota"', () => {
+    const spyLog = jest.spyOn(logger, 'error')
+    expect(IotaClientSingleton.getInstance()).toBeUndefined()
+    expect(spyLog).toHaveBeenCalledWith("couldn't connect to iota")
+  })
+})
```

Comment:
sollten die Methoden sendDataMessage() und getMessage nicht auch getestet werden?

Reply:
Wie denn?
Die beiden Funktionen sendDataMessage und getMessage sind Memberfunktionen der IotaClientSingleton Klasse.
Wenn getInstance undefined zurückgibt, also keine Instance von der Klasse IotaClientSingleton dann kann ich auch die Memberfunktionen der Klasse nicht aufrufen.

Reply:
Ah jetzt hab ich das geblickt...
Du hast ja zwei verschiedene Test-Klassen für den IotaClientSingleton gebaut
- eine die senden und empfangen testet
- eine die singleton disabled/misconfiguration testet

Klar dann geht kein testen der senden und empfangen Methoden. Sorry.
Da ergibt sich für mich die Frage, warum das in zwei verschiedene Testklassen und nicht in einer Testklasse verschiedene Testcases daraus zu machen? Aber da bin ich nicht der Experte...